clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ChannelResourceDefinition.java

```diff
@@ -62,24 +62,24 @@ public static PathElement pathElement(String name) {
     }
 
     public enum Attribute implements org.jboss.as.clustering.controller.Attribute {
-        STACK("stack", ModelType.STRING, null),
-        MODULE("module", ModelType.STRING, new ModelNode("org.wildfly.clustering.server"), new ModuleIdentifierValidatorBuilder()),
+        STACK("stack", ModelType.STRING, false, null), // 'stack' is required since model 4.0.0
+        MODULE("module", ModelType.STRING, true, new ModelNode("org.wildfly.clustering.server"), new ModuleIdentifierValidatorBuilder()),
         ;
         private final AttributeDefinition definition;
 
-        Attribute(String name, ModelType type, ModelNode defaultValue) {
-            this.definition = createBuilder(name, type, defaultValue).build();
+        Attribute(String name, ModelType type, boolean allowNull, ModelNode defaultValue) {
+            this.definition = createBuilder(name, type, allowNull, defaultValue).build();
         }
 
-        Attribute(String name, ModelType type, ModelNode defaultValue, ParameterValidatorBuilder validator) {
-            SimpleAttributeDefinitionBuilder builder = createBuilder(name, type, defaultValue);
+        Attribute(String name, ModelType type, boolean allowNull, ModelNode defaultValue, ParameterValidatorBuilder validator) {
+            SimpleAttributeDefinitionBuilder builder = createBuilder(name, type, allowNull, defaultValue);
             this.definition = builder.setValidator(validator.configure(builder).build()).build();
         }
 
-        private static SimpleAttributeDefinitionBuilder createBuilder(String name, ModelType type, ModelNode defaultValue) {
+        private static SimpleAttributeDefinitionBuilder createBuilder(String name, ModelType type, boolean allowNull, ModelNode defaultValue) {
```

Comment:
The allowNull parameter is unnecessary.  It can inferred via (defaultValue != null).